src/promise.rs

```diff
@@ -342,39 +418,44 @@ impl<T: Send + 'static> PromiseImpl<T> {
         }
     }
 
-    fn block_until_ready_mut(&mut self) -> &mut T {
+    fn block_until_ready_mut(&mut self, task_type: TaskType) -> &mut T {
+        // Constantly poll until we're ready.
+        while self.poll(task_type).is_pending() {
+            // Tick unless poll does it for us.
+            #[cfg(all(feature = "smol", not(feature = "tick-poll")))]
+            Self::tick(task_type);
+        }
         match self {
-            Self::Pending(rx) => {
-                let value = rx.recv().expect("The Promise Sender was dropped");
-                *self = Self::Ready(value);
-                match self {
-                    Self::Ready(ref mut value) => value,
-                    Self::Pending(_) => unreachable!(),
-                }
-            }
```

Comment:
Please don't remove the code that is optimal for non-smol backends!

Reply:
Sorry!